core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -789,8 +790,19 @@ private void loadMetadata(LockedInodePath inodePath, LoadMetadataContext context
               LOG.debug("Failed to loadMetadata because file is not in ufs:"
                       + " inodePath={}, options={}.",
                   childURI, loadMetadataContext, e);
+            } catch (BlockInfoException | FileAlreadyCompletedException
+                | FileDoesNotExistException | InvalidFileSizeException
+                | IOException e) {
+              LOG.debug("Failed to loadMetadata because the ufs file or directory"
+                      + " is {}, options={}.",
+                      childStatus, loadMetadataContext, e);
+              failedSync++;
             }
           }
+          if (failedSync > 0) {
+            throw new IOException("Failed to load metadata of " + failedSync
+                    + " files or directories.");
+          }
```

Comment:
If you throw IOE here, you are catching it on line 809 and re-throwing, which is unnecessary. I think you can throw outside the try-catch.

Also, more than just logging the failed count, can you also log the total and succeeded count?